quakesaver_client/cli.py

```diff
@@ -0,0 +1,77 @@
+import asyncio
+import logging
+from functools import wraps
+
+import click
+
+from quakesaver_client.sensor_actor import SensorActor
+
+logger = logging.getLogger("quakesaver_client")
+logging.basicConfig(level=logging.DEBUG)
+
+
+@click.option("--debug", is_flag=True, type=bool)
+@click.group()
+def cli(debug: bool):
+    if debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+
+def click_coro(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        return asyncio.run(f(*args, **kwargs))
+
+    return wrapper
+
+
+async def probe_sensor(host):
+    logger.debug(f"probing {host}")
+    sensor = SensorActor(host)
+    alive = await sensor.is_alive()
+    if alive:
+        state = await sensor.get_state()
+        uid = state["uid"]
+        version = state["software_version"]
+        logger.info(f"Sensor {uid}@{host} (version {version}) is alive")
+        return uid, host
+    else:
+        logger.debug("No running sensor software found on %s", host)
+
+
+@cli.command()
+@click.argument("hosts")
+@click_coro
+async def detect(hosts: str) -> None:
+    """Detect QuakeSaver sensors.
+
+    Args:
+        hosts (str): IP range to scan for sensors in CIDR notation.
+    """
+    logger.info(f"detecting hosts at {hosts}")
+    hosts_and_range = hosts.split("/")
+    fn_alive_sensors = "sensors-alive.csv"
+
+    assert len(hosts_and_range) <= 2
+    if len(hosts_and_range) == 1:
+        hosts = [hosts_and_range]
+
+    else:
+        host, ip_range = hosts_and_range
+        host = host.split(".")[0:3]
+        hosts = []
+        for i in range(1, 256):
+            hosts.append(f"{'.'.join(host)}.{i}")
+
```

Comment:
Use `IPv4Network` instead https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Network

```py
network = IPv4Network(hosts)

for ip in network:
    probe_sensor(ip)
```

also consider using `IPv4Address` instead of `str`.

Reply:
done